c7n/resources/iam.py

```diff
@@ -302,6 +302,21 @@ def _user_policies(resource):
         return matched
 
 
+@User.filter_registry.register('access-key-last-used')
+class UserAccessKeyLastUsed(Filter):
+    schema = type_schema(
+        'access-key-last-used',
+        keyid={'type': 'string'})
+
+    def process(self, resources, event=None):
+        keyid = self.data.get('keyid', None)
```

Comment:
Yeah, I had actually started rethinking this after I submitted the PR... realized that it would be better to actually make it filter from a range and not by keyid.